R/tar_flood_exposure.R

```diff
@@ -67,13 +69,31 @@ zonal_pop_exposure <- function(floodscan_path=fp_fs,
      )
  }
   if(!binarize_floodscan){
-    if(length(flood_frac_thresh)>1){
```

Comment:
you should include this purrr::map iteration in the binarize_floodscan section as well if you really want the binary thresholds to be implemented as suggested in the target name `*_stats_binary`.

In the section where it's not making a binary raster:

```
if(!binarize_floodscan) {
... 
} 
```
all it's doing is doing is setting values < theshold to 0 and retaining original raw values above that. Therefore, you would want to add this iteration step to the section that is doing the binary reclassification

```
if(binarize_floodscan) {
 ...
}
```

I think what could have happened is that I started to write this `map()` loop at some point during the Somalia risk work, but we never ended up using it because we didn't test various thresholds, rather just went with 20 % flood fraction.

Reply:
so the pseudo code would be like:

```
if(binarize_floodscan){
    if (length(flood_frac_thresh)>1){
        flood_frac_thresh |>
            map( \(thresh_tmp){
                r_binary_tmp  <- ifel(r_fs_resampled>=thresh_tmp,1,0) 
                r_exposure <- r_binary_tmp * r_wp

               # now run zonal stats on R exposure

                })
}
               
# then for not binarizing (if you even want this)

if(!binarize_floodscan){
    if (length(flood_frac_thresh)>1){              
       flood_frac_thresh |>
            map( \(thresh_tmp){
                r_thresholded_tmp  <- ifel(r_fs_resampled>=thresh_tmp,r_fs_resampled,0) 
                r_exposure <- r_thresholded_tmp * r_wp 
    # etc. etc.


  lgl_thresh_mask <- ifel(r_fs_resampled>=flood_frac_thresh,1,0) 
   r_fs_resampled_masked <- mask(r_fs_resampled,lgl_thresh_mask)
   # r_exposure <- r_fs_resampled_masked * r_wp
   r_exposure <- lgl_thresh_mask * r_wp
   ... 

  zonal_stats ... 
}
```